app/src/forms/auth/middleware/params.js

```diff
@@ -0,0 +1,92 @@
+const Problem = require('api-problem');
+const uuid = require('uuid');
+
+const formService = require('../../form/service');
+
+/**
+ * Validates that the :formId route parameter exists and is a UUID.
+ *
+ * @param {*} _req the Express object representing the HTTP request - unused
+ * @param {*} _res the Express object representing the HTTP response - unused
+ * @param {*} next the Express chaining function
+ * @param {*} formId the :formId value from the route
+ */
+const validateFormId = async (_req, _res, next, formId) => {
+  try {
+    if (!uuid.validate(formId)) {
+      throw new Problem(400, {
+        detail: 'Bad formId',
+      });
+    }
+
+    next();
+  } catch (error) {
+    next(error);
+  }
+};
+
+/**
+ * Validates that the :formVersionDraftId route parameter exists and is a UUID.
+ * This validator requires that the :formId route parameter also exists.
+ *
+ * @param {*} req the Express object representing the HTTP request
+ * @param {*} _res the Express object representing the HTTP response - unused
+ * @param {*} next the Express chaining function
+ * @param {*} formVersionDraftId the :formVersionDraftId value from the route
+ */
+const validateFormVersionDraftId = async (req, _res, next, formVersionDraftId) => {
+  try {
+    if (!uuid.validate(formVersionDraftId)) {
+      throw new Problem(400, {
+        detail: 'Bad formVersionDraftId',
+      });
+    }
+
+    const formVersionDraft = await formService.readDraft(formVersionDraftId);
```

Comment:
This service call does add another database query to these routes, and the same in `validateFormVersionId` below.

We could think about doing some some of caching of these values once they're pulled from the database, but that would make things more complicated. Plus we'd have to deal with cache invalidation and things like that. Maybe keep it simple for now and do optimization later if we need it.

Ideally the database calls should be small and quick. The database itself should be caching these things and do a better job of it than we would!

Reply:
Am I missing something here? (I haven't run this). For draft and form version don't you need the form id too to validate if they belong to each other?

Reply:
offline discussion cleared this up. early morning reading comprehension must be low.